File: 2025-06-June-GFG-POTD/June 11 - Remove the balls/Solution.py
from typing import List

class Solution:
    def findLength(self, color: List[int], radius: List[int]) -> int:
        stack = []

        for c, r in zip(color, radius):
            if stack and stack[-1] == (c, r):
                stack.pop()  # Remove matching pair
            else:
                stack.append((c, r))

        return len(stack)  # Remaining balls

File: 2025-06-June-GFG-POTD/June 11 - Remove the balls/Solution.cpp
#include <vector>
#include <stack>
using namespace std;

class Solution {
public:
    int findLength(vector<int>& color, vector<int>& radius) {
        stack<pair<int, int>> stk;

        for (int i = 0; i < color.size(); i++) {
            if (!stk.empty() && stk.top().first == color[i] && stk.top().second == radius[i]) {
                stk.pop(); // Remove matching pair
            } else {
                stk.push({color[i], radius[i]});
            }
        }

        return stk.size(); // Remaining balls
    }
};
